File: packages/shared-types/src/entities/Reaction.ts
import { ReactionType } from '@commentable/shared-enums';
import { BaseEntity } from './BaseEntity';

/**
 * Reaction Entity
 *
 * Represents a user's reaction to a comment
 */
export interface Reaction extends BaseEntity {
  userId: string;
  commentId: string;
  type: ReactionType;
}

/**
 * Reaction summary for a comment
 */
export interface ReactionSummary {
  commentId: string;
  reactions: {
    type: ReactionType;
    count: number;
  }[];
  totalCount: number;
  userReaction?: ReactionType | null; // Current user's reaction
}

File: apps/api/Domain/Entities/Reaction.cs
using CommentableAPI.Domain.Enums;

namespace CommentableAPI.Domain.Entities;

/// <summary>
/// Reaction entity representing user reactions to comments
/// </summary>
public class Reaction : BaseEntity
{
    /// <summary>
    /// Foreign key to the user who created the reaction
    /// </summary>
    public Guid UserId { get; set; }

    /// <summary>
    /// Foreign key to the comment being reacted to
    /// </summary>
    public Guid CommentId { get; set; }

    /// <summary>
    /// Type of reaction (Like, Dislike, Love, Clap, Laugh, Sad)
    /// </summary>
    public ReactionType ReactionType { get; set; }

    // Navigation properties

    /// <summary>
    /// The user who created this reaction
    /// </summary>
    public User User { get; set; } = null!;

    /// <summary>
    /// The comment this reaction belongs to
    /// </summary>
    public Comment Comment { get; set; } = null!;
}
